test/unittests/commands/xmlupload/test_ontology_client.py

```diff
@@ -0,0 +1,245 @@
+        },
+    ]
+    res_prop = _get_all_properties_from_graph(test_graph)
+    assert res_prop == ["testonto:hasResource", "testonto:hasResourceValue"]
+
+
+def test_get_all_properties_from_graph_resources_and_properties() -> None:
+    test_graph = [
+        {
+            "knora-api:isResourceClass": True,
+            "rdfs:label": "Sequenz einer Audio-Ressource",
+            "knora-api:canBeInstantiated": True,
+            "rdfs:subClassOf": [],
+            "@type": "owl:Class",
+            "@id": "testonto:AudioSequence",
+        },
+        {
+            "rdfs:label": "URI",
+            "rdfs:subPropertyOf": {},
+            "knora-api:isEditable": True,
+            "knora-api:isResourceProperty": True,
+            "@type": "owl:ObjectProperty",
+            "salsah-gui:guiAttribute": [],
+            "knora-api:objectType": {},
+            "salsah-gui:guiElement": {},
+            "@id": "testonto:hasUri",
+        },
+    ]
+    res_prop = _get_all_properties_from_graph(test_graph)
+    assert res_prop == ["testonto:hasUri"]
+
+
+def test_deserialize_ontology() -> None:
+    test_graph = [
+        {
+            "knora-api:isResourceClass": True,
+            "rdfs:label": "Annotation",
+            "knora-api:canBeInstantiated": True,
+            "rdfs:subClassOf": [],
+            "rdfs:comment": "A generic class for representing annotations",
+            "@type": "owl:Class",
+            "@id": "knora-api:Annotation",
+        },
+        {
+            "rdfs:label": "has Link to",
+            "rdfs:subPropertyOf": {},
+            "knora-api:isEditable": True,
+            "knora-api:isResourceProperty": True,
+            "@type": "owl:ObjectProperty",
+            "knora-api:objectType": {},
+            "@id": "knora-api:hasLinkTo",
+            "knora-api:subjectType": {},
+            "knora-api:isLinkProperty": True,
+            "rdfs:comment": "Represents a direct connection between two resources",
+        },
+    ]
+    res_onto = deserialize_ontology(test_graph)
+    assert res_onto.classes == ["Annotation"]
+    assert res_onto.properties == ["hasLinkTo"]
+
+
+def test_get_ontology_names_from_server() -> None:
+    response = {
+        "project": {
+            "description": [],
+            "id": "http://rdfh.ch/projects/yTerZGyxjZVqFMNNKXCDPF",
+            "keywords": [],
+            "logo": None,
+            "longname": "Bernoulli-Euler Online",
+            "ontologies": [
+                "http://www.knora.org/ontology/0801/biblio",
+                "http://www.knora.org/ontology/0801/newton",
+                "http://www.knora.org/ontology/0801/leibniz",
+                "http://www.knora.org/ontology/0801/beol",
+            ],
+            "selfjoin": False,
+            "shortcode": "0801",
+            "shortname": "beol",
+            "status": True,
+        }
+    }
+    con = ConnectionMock(response)
+    onto_cli = OntologyClientLive(con, "0801", "beol", Path(""))
```

Comment:
the abbreviation "cli" stands for Command Line Interface. So I wouldn't abbreviate at all, just write `onto_client`